### DISCUSSION TITLE
MPC build issues - 4.2.2 using Bugfix 2.1.x

### DISCUSSION THREAD

**Discussion by Foowy** [mriscoc/Ender3V2S1#391] [Q&A] [answered]
Build issues when building MPC chosen for firmware. 

System/Board : 3V2 4.2.2
Marlin Code: Bugfix 4.1.x
Hotend: Sprite Pro
Configurator tool usage: Ender3V2-422-BLTUBL-T13-MPC

errors:

Marlin\src\lcd\e3v2\proui\dwin.cpp: In function 'void ApplyPIDi()':
Marlin\src\lcd\e3v2\proui\dwin.cpp:2683:20: error: 'class Temperature' has no member named 'updatePID'
 2683 |     thermalManager.updatePID();
      |                    ^~~~~~~~~
Marlin\src\lcd\e3v2\proui\dwin.cpp: In function 'void ApplyPIDd()':
Marlin\src\lcd\e3v2\proui\dwin.cpp:2687:20: error: 'class Temperature' has no member named 'updatePID'
 2687 |     thermalManager.updatePID();
      |                    ^~~~~~~~~
Compiling .pio\build\STM32F103RC_creality\src\src\libs\hex_print.cpp.o
Compiling .pio\build\STM32F103RC_creality\src\src\libs\least_squares_fit.cpp.o
Compiling .pio\build\STM32F103RC_creality\src\src\libs\nozzle.cpp.o
*** [.pio\build\STM32F103RC_creality\src\src\lcd\e3v2\proui\dwin.cpp.o] Error 1

Apologies if this should be over on Marlin main, but error is relative to proui. I had seen the error initially when trying to build MPC and LA together, removed LA from configurator, build error is the same at just MPC. Tested with T1 and T13 to rule out thermistor values.

**Comment by mriscoc** [comment, marked as answer]
Use an older configuration files whose dates match the date of the firmware release sources or wait to the next release.

**Comment by Foowy** [comment]
This is an update for Mriscoc and others,

Capable of running ender 3V2 with sprite pro on 4.2.2 board with T13, LA, UBL, MPC without any hitches or slowdowns. No issues with printing and level, Hot-end temps did not fluctuate.

Build was done with git clone of Mriscoc and moving over special configuration change files. Due to release timing you will need to manually modify configuration files from #define CONFIGURATION_H_VERSION 02010200 to #define CONFIGURATION_H_VERSION 02010100 to get it to compile.. the key change is 02010100.

I did turn off power loss protection as referenced in another post as that appears to cause the hitching/pausing problem in the V2 systems. I have now just been turning it off by default as it fixed the hitching problems i had seen before. 

I have not tested with it on, but presume that hitching will return if enabled.

Side note, since i don't understand GitHub and its dating on comments, today is 10/1/2022 (US) or 01/10/2022 (almost everyone else)
